src/librustdoc/html/static/js/search.js

```diff
@@ -15,6 +15,25 @@ if (!Array.prototype.toSpliced) {
     };
 }
 
+function takeFromArray(array, count) {
+    const result = [];
+    while (count > 0) {
+        result.push(array.shift());
+        count -= 1;
+    }
+    return result;
+}
```

Comment:
```suggestion
function takeFromArray(array, count) {
    return array.splice(0, count);
}
```

Is at least twice faster based on https://jsbench.me/.

Reply:
Sounds good to me as well!